code/datums/abilities/critter/drop_disguise.dm

```diff
@@ -0,0 +1,26 @@
+/datum/targetable/critter/drop_disguise
+	name = "Drop Disguise"
+	desc = "Shed your skin and return to your base form."
+	icon_state = "drop_disguise"
+	cooldown = 3 SECONDS
+	targeted = 0
+
+	cast()
+		if (..())
+			return TRUE
+		var/mob/living/critter/mimic/user = holder.owner
+		if (user.base_form)
+			boutput(holder.owner, SPAN_ALERT("You're in your base form already!"))
+			return TRUE
+		SETUP_GENERIC_PRIVATE_ACTIONBAR(user, null, 2 SECONDS, /datum/targetable/critter/drop_disguise/proc/drop, user, null, null, null, INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_ACTION)
```

Comment:
This should either be instant or not interrupted by movement, you're going to want to do this most often when you're being found out and having a movement interrupted actionbar is going to get you killed.